### PR TITLE
Only members register itself as a service into ZooKeeper

### PR DESCRIPTION
**Pull request by jerrinot**
There is no need to register clients. Nobody cares about theirs
existence.

### CODE REVIEW

**Comment by jerrinot** [comment]
@emre-aydin: good finding. 
I fixed the NPE and added at least a very basic test with a client.

**Comment by hasancelik** [comment]
verify

File: pom.xml [hunk 1]
```diff
@@ -170,7 +171,6 @@
                         -Xms128m -Xmx1G -XX:MaxPermSize=128M
                         -Dhazelcast.phone.home.enabled=false
                         -Dhazelcast.mancenter.enabled=false
-                        -Dhazelcast.logging.type=none
```
Comment (jerrinot) [inline pom.xml]: I am not sure why logging was disabled in the first place.

File: pom.xml [hunk 2]
```diff
@@ -233,6 +233,43 @@
             <scope>test</scope>
         </dependency>
 
+        <dependency>
```
Comment (jerrinot) [inline pom.xml]: all newly added dependencies are in the `test` scope

File: pom.xml [hunk 3]
```diff
@@ -233,6 +233,43 @@
...
+            <groupId>com.hazelcast</groupId>
+            <artifactId>hazelcast</artifactId>
+            <version>${hazelcast.version}</version>
+            <classifier>tests</classifier>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>org.apache.logging.log4j</groupId>
```
Comment (jerrinot) [inline pom.xml]: we use log4j2 in all tests.

File: pom.xml [hunk 4]
```diff
@@ -233,6 +233,43 @@
...
+            <artifactId>log4j-core</artifactId>
+            <version>${log4j2.version}</version>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>org.apache.logging.log4j</groupId>
```
Comment (jerrinot) [inline pom.xml]: Curator framework uses slf4j for logging. so let's add a bridge slf4j -> log4j2

**Comment by noctarius** [comment]
Looks good from my side.

**Pull request merged**